modules/ding_carousel/js/ding_carousel.js

```diff
@@ -0,0 +1,303 @@
+        // the material ids and coverts them into ding_availability format and
+        // updates the settings, which is this used when behaviors are attached
+        // below. This is a hack, but the alternative was to re-write
+        // ding_availability.
+        var matches = data.content.match(/reservation-\d+-\w+:\d+/gm);
+        if (matches instanceof Array) {
+          if (!Drupal.settings.hasOwnProperty('ding_availability')) {
+            Drupal.settings.ding_availability = {};
+          }
+          for (var i in matches) {
+            var match = matches[i];
+            var id = match.substring(match.indexOf(':') + 1);
+            match = match.replace('reservation', 'availability').replace(':', '');
+            Drupal.settings.ding_availability[match] = [ id ];
+          }
+        }
+
+        // Ensure that behaviors are attached to the new content.
+        Drupal.attachBehaviors($('.ding-carousel-item'));
+
+        // Carry on processing the queue.
+        running = false;
+        update();
+      }
+    });
+  };
+
+  /**
+   * Event handler for progressively loading more covers.
+   */
+  var update_handler = function (e, slick) {
+    var tab = e.data;
+
+    if (!tab.data('updating')) {
+      // If its the first batch or we're near the end.
+      if (tab.data('offset') === 0 ||
+          (tab.data('offset') > -1 &&
+           (slick.slideCount - slick.currentSlide) <
+           (slick.options.slidesToScroll * 2))) {
+        // Disable updates while updating.
+        tab.data('updating', true);
+        // Add to queue.
+        queue.push({'tab': tab, 'slick': slick, 'target': $(e.target)});
+      }
+    }
+    // Run queue.
+    update();
+  };
+
+  /**
+   * Fudge slidesToScroll for slick.
+   *
+   * In variableWith mode, we don't know how many slides are shown, so
+   * we can't set slidesToScroll to any sane value apart from 1, which
+   * isn't cool. So we calculate it here.
+   *
+   * Obviously this can more or less fail if the first slide has an
+   * odd width, but we expect all slides to be pretty similar.
+   */
+  var update_slides_to_scroll = function (e, slick) {
+    var slidesToScroll = Math.floor(slick.$slider.width() / slick.$slides.eq(0).outerWidth(true)) - 1;
+    slick.options.slidesToScroll = Math.max(slidesToScroll, 1);
+    slick.options.slidesToShow = slick.$slider.width() / slick.$slides.eq(0).outerWidth(true);
+  };
+
+  /**
+   * Start the carousel when the document is ready.
+   */
+  Drupal.behaviors.ding_carousel = {
+    attach: function (context) {
+      // Start all carousels, tabbed or standalone.
+      $('.ding-carousel', context).each(function () {
+        var carousel = $(this);
+
+        var settings = {};
+        if (typeof $(this).data('settings') === 'object') {
+          settings = $(this).data('settings');
+        }
+
+        // Reset ul to 100% width before we start Slick.
```

Comment:
It's from @xendk original PR, so maybe he knows. But it might be something to do with slick and variableWidth to allow the theme to set the width of the slids.

Reply:
:+1: